multidim_threshold/search.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import multidim_threshold
+from collections import namedtuple
+import more_itertools
+
+project_along = namedtuple('project_along', 'root direc')
+
+
+new_hi = lambda pi, hi: pi.root + np.multiply(pi.direc, np.true_divide(np.array(hi) - np.array(pi.root), np.array(pi.direc)).min())
+midpoint = lambda a, b: np.true_divide(np.array(a) + np.array(b), 2)
+
+def project_along_axes(lo, mid):
+    return [project_along_i(lo[:], mid, i) for i in range(len(lo))]
+
+def project_along_i(lo, mid, i):
+    lo[i] = mid[i]
+    return lo
+
+
+def project_singleLambda(hi, is_member, pi):
+    new_high = new_hi(pi, hi)
+    rec = multidim_threshold.to_rec(pi.root, new_high)
+    map = {}
```

Comment:
probably best not to overwrite the `map` keyword.  Based on usage seems to be a map of trace id to the result of the projection right? maybe call it `proj`?